### PR TITLE
Reduce again the number of match calls

### PR DESCRIPTION
**Pull request by Hofer-Julian**
- Reduce the number of match calls (see commit messages)
- A few miscellaneous improvements

### CODE REVIEW

**Comment by Hofer-Julian** [comment]
I've had another close look at the changes and arranged the commits into meaningful pieces.
The PR is now ready to be reviewed from my side.

This is a big changeset and the test suite isn't foolproof, so feel free to point out anything that you are not sure about.

File: crates/nu-cli/src/prompt_update.rs [hunk 1]
```diff
@@ -107,12 +106,13 @@ pub(crate) fn update_prompt<'prompt>(
...
-            Some(prompt_string) => Some(format!(
+        if let Some(prompt_string) = left_prompt_string {
+            Some(format!(
                 "{}{}{}",
                 PRE_PROMPT_MARKER, prompt_string, POST_PROMPT_MARKER
-            )),
-            None => left_prompt_string,
+            ))
+        } else {
+            left_prompt_string
```
Comment (sholderbach) [inline crates/nu-cli/src/prompt_update.rs:115]: I think in this case `.map`/`.and_then` would actually be less confusing.

File: crates/nu-cli/src/reedline_config.rs [hunk 2]
```diff
@@ -651,13 +651,14 @@ fn add_parsed_keybinding(
             let pos1 = char_iter.next();
             let pos2 = char_iter.next();
 
-            let char = match (pos1, pos2) {
-                (Some(char), None) => Ok(char),
-                _ => Err(ShellError::UnsupportedConfigValue(
+            let char = if let (Some(char), None) = (pos1, pos2) {
+                Ok(char)
+            } else {
+                Err(ShellError::UnsupportedConfigValue(
```
Comment (sholderbach) [inline crates/nu-cli/src/reedline_config.rs:657]: Here the wrapping in `Ok` reads a bit weird as the `Err` is immediately bubbled up on L602 with a `?;`

File: crates/nu-command/src/database/values/sqlite.rs [hunk 3]
```diff
@@ -107,12 +107,7 @@ impl SQLiteDatabase {
     }
 
     pub fn open_connection(&self) -> Result<Connection, rusqlite::Error> {
-        let conn = match Connection::open(&self.path) {
-            Ok(conn) => conn,
-            Err(err) => return Err(err),
-        };
-
-        Ok(conn)
```
Comment (sholderbach) [inline crates/nu-command/src/database/values/sqlite.rs:115]: LOL

File: crates/nu-protocol/src/config.rs [hunk 4]
```diff
@@ -1548,18 +1537,21 @@ fn create_hooks(value: &Value) -> Result<Hooks, ShellError> {
...
+                    span,
+                ))
+            } else {
+                Err(ShellError::UnsupportedConfigValue(
+                    "record for 'hooks' config".into(),
+                    "non-record value".into(),
+                    Span::unknown(),
+                ))
+            }
+        }
```
Comment (sholderbach) [inline crates/nu-protocol/src/config.rs:1554]: I think this is a very complicated way of saying just `Err(...., v.span().unwrap_or_else(Span::unknown))`

**Review by sholderbach — COMMENTED** [review]
Thanks for this tour de force. Got till have way through `nu-protocol`/[e05640](https://github.com/nushell/nushell/pull/7815/commits/e056408569878abd6d82bf95a7d5942bbd02d557) tonight. So many places that clean up a bit nicer.
I personally am not in the full `if let Some() = ...` vs a two arm match camp when both arms are warranted but the reduced indentation is definitely a plus.

File: crates/nu-cli/src/prompt_update.rs [hunk 5]
```diff
@@ -77,19 +77,18 @@ fn get_prompt_string(
...
+                    x.pop();
                 }
-                None => None,
+
+                if x.ends_with('\r') {
+                    x.pop();
+                }
+                Some(x)
+            } else {
+                None
```
[the remark this answers is not in the document]
Comment (Hofer-Julian) [reply crates/nu-cli/src/prompt_update.rs]: It does in my opinion.
Changed.

File: crates/nu-cli/src/reedline_config.rs (hunk 2 above)
Comment (Hofer-Julian) [reply crates/nu-cli/src/reedline_config.rs]: Good point.
Fixed.

File: crates/nu-protocol/src/config.rs (hunk 4 above)
Comment (Hofer-Julian) [reply crates/nu-protocol/src/config.rs]: That is indeed nicer.
Fixed.

File: crates/nu-cli/src/prompt_update.rs (hunk 1 above)
Comment (Hofer-Julian) [reply crates/nu-cli/src/prompt_update.rs:113]: Agreed and changed.

**Review by Hofer-Julian — COMMENTED** [review]
> I personally am not in the full if let Some() = ... vs a two arm match camp when both arms are warranted but the reduced indentation is definitely a plus.

I actually agree that they are not always an improvement.
I had many more in the first iteration, but especially when the output is assigned to a value, they can be less readable than `match`.

**Comment by sholderbach** [comment]
Thanks. I think we are good here!

**Pull request merged**